src/main/java/edu/cornell/kfs/module/ar/service/impl/CuAREmailServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package edu.cornell.kfs.module.ar.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.module.ar.ArKeyConstants;
+import org.kuali.kfs.module.ar.businessobject.CustomerAddress;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.service.impl.AREmailServiceImpl;
+
+import java.text.MessageFormat;
+
+public class CuAREmailServiceImpl extends AREmailServiceImpl {
+    
+    @Override
+    protected String getSubject(ContractsGrantsInvoiceDocument invoice) {
+        String grantNumber = invoice.getInvoiceGeneralDetail().getAward().getProposal().getGrantNumber();
+        String subject;
+        String message;
+        if (StringUtils.isBlank(grantNumber)) {
+            subject = kualiConfigurationService.getPropertyValueAsString(ArKeyConstants.CGINVOICE_EMAIL_SUBJECT_NO_GRANT_NUMBER);
+            message = MessageFormat.format(subject, invoice.getDocumentNumber(), 
+                    invoice.getInvoiceGeneralDetail().getAward().getProposal().getProposalNumber());
+        } else {
+            subject = kualiConfigurationService.getPropertyValueAsString(ArKeyConstants.CGINVOICE_EMAIL_SUBJECT);
+            message = MessageFormat.format(subject, invoice.getDocumentNumber(), 
+                    invoice.getInvoiceGeneralDetail().getAward().getProposal().getGrantNumber(), 
+                    invoice.getInvoiceGeneralDetail().getAward().getProposal().getProposalNumber());
```

Comment:
Another spot where the proposal number can be read from the InvoiceGeneralDetail instead.

Reply:
Same issue here. Do we really want to deviate from base code even further?

Reply:
Similarly to the other spot, deriving the value like this is fine with me, and will leave it as-is.